gobblin-data-management/src/main/java/gobblin/data/management/copy/extractor/FileAwareInputStreamExtractor.java

```diff
@@ -62,10 +62,8 @@ public FileAwareInputStream readRecord(@Deprecated FileAwareInputStream reuse)
       throws DataRecordException, IOException {
 
     if (!this.recordRead) {
-      Configuration conf = HadoopUtils.newConfiguration();
-      FileSystem fsFromFile = this.file.getOrigin().getPath().getFileSystem(conf);
       this.recordRead = true;
-      return new FileAwareInputStream(this.file, fsFromFile.open(this.file.getFileStatus().getPath()));
+      return new FileAwareInputStream(this.file, this.fs.open(this.file.getFileStatus().getPath()));
```

Comment:
This change was introduced in #1306 and broke lumos-ng puller. The source filesystem cannot be got from an empty `configuration`. 

I was unaware of the multiple `FileSystem` mode. One possible solution may be instead of using `path.getFs`, will `copyfile.getOriginFs` work for the multiple `FileSystem` mode? Another solution could be creating one more constructor for `FileAwareInputStreamExtractor`, which can pass the job `properties`/`state` in. Then `path.getFs` can have a `FileSystem` instance for `http` uri.